src/models/model.ts

```diff
@@ -0,0 +1,44 @@
+import { Attribute } from './attributes';
+
+export type SaveCallback = (error: Error | null, result?: any) => void;
+
+export default class Model {
+  static attributes: { [key: string]: Attribute };
```

Comment:
I was toying with the idea of abstracting classes and even variables like this one but I ran into some issues:

- when making `Model` abstract and you try to create an instance of class `typeof Model` typescript throws an error "Cannot create an instance of an abstract class" even though you are trying to instantiate a class `typeof Model`, not directly of `Model`
- Making `attributes` static means making `Model` static  and every class extending `Model` must implement their own `attributes`, but the implementation would all be the same so it would be a duplicate implementation across all classes that extend `Model`

So given the added complexity of making it abstract I decided against it